service/common/src/test/java/org/apache/polaris/service/storage/StorageConfigurationTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.storage;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.*;
+
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.GoogleCredentials;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.function.Supplier;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.services.sts.StsClient;
+
+public class StorageConfigurationTest {
+
+  private static final String TEST_ACCESS_KEY = "test-access-key";
+  private static final String TEST_GCP_TOKEN = "ya29.test-token";
+  private static final String TEST_REGION = "us-west-2";
+  private static final String TEST_SECRET_KEY = "test-secret-key";
+  private static final Duration TEST_TOKEN_LIFESPAN = Duration.ofMinutes(20);
+
+  @BeforeEach
+  public void setUpAwsRegion() {
+    System.setProperty("aws.region", TEST_REGION);
```

Comment:
This is risky in case tests are executed concurrently. Do we absolutely have to use system properties for this test?

The test already uses static mocks... Could we mock the code that need this setting?

Reply:
I can try to mock it for sure. This wasn't failing locally so I wanted to check if this worked. Let me get this mocked in a follow up commit

Reply:
@dimas-b might need to approval to run the workflow again